pkg/config/config_template.yaml

```diff
@@ -430,429 +787,405 @@ api_key:
-# about the docker daemon load.
+#  exclude_pause_container: true
+
+## @param docker_query_timeout - integer - optional - default: 5
+## Set the default timeout value when connecting to the docker daemon.
 #
-# exclude_pause_container: true
+#  docker_query_timeout: 5
 
-# Exclude default containers from DockerCloud:
-# The following configuration will instruct the agent to ignore the containers from Docker Cloud.
-# You can remove the ones you want to collect.
-# ac_exclude: ["image:dockercloud/network-daemon","image:dockercloud/cleanup","image:dockercloud/logrotate","image:dockercloud/events","image:dockercloud/ntpd"]
-# ac_include: []
+## @param ad_config_poll_interval - integer - optional - default: 10
+## The default interval in second to check for new autodiscovery configurations
+## on all registered configuration providers.
 #
-# You can also use the regex to ignore them all:
-# ac_exclude: ["image:dockercloud/*"]
-# ac_include: []
+#  ad_config_poll_interval: 10
+
+{{ end -}}
+{{- if .ClusterChecks }}
+
+#################################
+## Cluster check Configuration ##
+#################################
+
+## @param cluster_checks - custom object - optional
+## Enter specific configurations for your cluster check.
+## The cluster-agent is able to autodiscover cluster resources and dispatch checks on
+## the node-agents (provided the clustercheck config provider is enabled on them).
+## Uncomment this parameter and the one below to enable them.
+## See https://docs.datadoghq.com/agent/kubernetes/cluster/
+#
+#  cluster_checks:
+
+  ## @param enabled - boolean - optional - default: false
+  ## Set to true to enable the dispatching logic on the leader cluster-agent.
+  #
+  #  enabled: false
+
+  ## @param node_expiration_timeout - integer - optional - default: 30
+  ## Set "node_expiration_timeout" time in second after which Node-agents that have not
+  ## queried the cluster-agent are deleted, and their checks re-dispatched to other nodes.
+  #
+  #  node_expiration_timeout: 30
+
+  ## @param warmup_duration - integer - optional - default: 30
+  ## Set the "warmup_duration" duration in second for the cluster-agent to wait for all
+  ## node-agents to report to it before dispatching configurations.
+  #
+  #  warmup_duration: 30
+
+  ## @param cluster_tag_name - string - optional - default: cluster_name
+  ## If a cluster_name value is set or autodetected, a "<CLUSTER_NAME>" tag is added
+  ## to all cluster-check configurations sent to the node-agents.
+  ## Set a custom tag name here, or disable it by setting an empty name.
+  #
+  #  cluster_tag_name: cluster_name
+
+  ## @param extra_tags - list of key:value elements - optional
+  ## Set a list of additionnal tags can to be added to every cluster-check configuration.
+  #
+  #  extra_tags:
+  #    - <TAG_KEY>:<TAG_VALUE>
+
+{{ end -}}
+{{- if .DockerTagging }}
+
+#########################
+## Container detection ##
```

Comment:
Shouldn't we just have a big Container section? instead of splitting it into several sections which make it difficult for the users to understand everything that is possible related to containers.

Also we should highlight in such sections what is automatically done when running the agent as a container.